GenderPayGap.WebUI/Controllers/Admin/AdminDatabaseIntegrityChecksController.cs

```diff
@@ -0,0 +1,201 @@
+                }
+            }
+
+            return PartialView("OrganisationsWithMultipleActiveScopesForASingleYear", model);
+        }
+
+        [HttpGet("organisations-with-no-active-scope-for-every-year-ajax")]
+        public async Task<IActionResult> OrganisationsWithNoActiveScopeForEveryYearAjax()
+        {
+            var model = new DatabaseIntegrityChecksViewModel();
+            List<Organisation> organisations = dataRepository.GetAll<Organisation>()
+                .Include(o => o.OrganisationScopes)
+                .ToList();
+
+            foreach (Organisation organisation in organisations)
+            {
+                DateTime organisationCreationDate = organisation.Created;
+                DateTime accountingDateForCreationYear = organisation.SectorType.GetAccountingStartDate(organisationCreationDate.Year);
+
+                int firstRequiredAccountingDateYear = organisationCreationDate < accountingDateForCreationYear
+                    ? accountingDateForCreationYear.AddYears(-1).Year
+                    : accountingDateForCreationYear.Year;
+
+                int latestRequiredAccountingDateYear = organisation.SectorType.GetAccountingStartDate().Year;
+
+                List<int> requiredYears = Enumerable.Range(
+                        firstRequiredAccountingDateYear,
+                        (latestRequiredAccountingDateYear - firstRequiredAccountingDateYear) + 1)
+                    .ToList();
+
+                List<int> yearsWithActiveScopes = organisation.OrganisationScopes
+                    .Where(os => os.Status == ScopeRowStatuses.Active)
+                    .GroupBy(scope => scope.SnapshotDate)
+                    .Select(g => g.Key.Year)
+                    .ToList();
+
+                foreach (int year in requiredYears)
+                {
+                    if (!yearsWithActiveScopes.Contains(year))
+                    {
+                        model.OrganisationsWithNoActiveScopeForEveryYear.Add(organisation);
+                    }
+                }
+            }
+
+            return PartialView("OrganisationsWithNoActiveScopeForEveryYear", model);
+        }
+
+        [HttpGet("organisations-with-multiple-active-returns-for-a-single-year-ajax")]
+        public async Task<IActionResult> OrganisationsWithMultipleActiveReturnsForASingleYearAjax()
+        {
+            var model = new DatabaseIntegrityChecksViewModel();
+            List<Organisation> organisations = dataRepository.GetAll<Organisation>()
+                .Include(o => o.Returns)
+                .ToList();
+
+            foreach (Organisation organisation in organisations)
+            {
+                List<Return> activeReturns = organisation.Returns
+                    .Where(r => r.Status == ReturnStatuses.Submitted)
+                    .ToList();
+
+                bool multipleActiveReturnsForYear = activeReturns
+                    .GroupBy(r => r.AccountingDate)
```

Comment:
Maybe even `.GroupBy(r => r.AccountingDate.Year)` to catch the case where there's an organisation with:
* 1 return for the public-sector date
* 1 return for the private-sector date

Reply:
Ooo, that's a very good point. I've added that in now